script/testing/junit/src/TracefileTest.java

```diff
@@ -184,14 +229,18 @@ public static void checkResultMatch(String hash1, String hash2, String message,
                 List<String> new_res = new ArrayList<>();
                 for(String i:res){
                     try{
-                        int cur = (int)Double.parseDouble(i);
+                        int cur = (int)Math.round(Double.parseDouble(i));
```

Comment:
This seems like a hack. What are you trying to do here?

Reply:
I was trying to deal with the float precision. If the hashes don't match but the result length are the same, I converted doubles from the result list to int and compared them again.

Reply:
I think specifying a delta to account for this precision drifting might be better than merely rounding.

Reply:
Remove the rounding.

Reply:
So this is for the case that hash don't match but result list size match. I did a work around here to cast doubles to int and then compare the hash again. How should I deal with floating difference without the rounding? We might only get the hash from the trace instead of exact results.

Reply:
For now people will have to modify the tracefile by hand to add the value list.

Reply:
I see. If a value list is used (instead of hash), we won't enter this piece of code. Instead, the checkEqual method will be called (the other unresolved comment)

Reply:
For clarification here, I asked Jeff to support this precision aware comparison (but yeah, rounding might not be the best way). 
Quick question here, If we need to enter the value by ourselves, is the trace test for regression test only? 
As we are simply comparing our noisepage result (ran in the JUnit test) with the noisepage results(manually generated and inputed from manual runs of same trace SQLs).